### PR TITLE
feat: show version number on homepage

### LINKED ISSUE #148: Add version number to homepage
image

### PR DESCRIPTION
**Pull request by saihaj**
### Summary of PR
Read version number from local package and show it on homepage.

### Tests for unexpected behavior

Kapture 2020-06-17 at 15 55 04

**Updated Screenshot** image

### Time spent on PR
1.5 hours

### Linked issues
Fix #148 

### Reviewers
@Harjot1Singh @bhajneet

### CODE REVIEW

**Comment by bhajneet** [comment]
Bold font weight doesn't match rest of page, places too much emphasis on the version number. Perhaps lower the opacity as well.

**Comment by saihaj** [comment]
Does this look good? 
image
image

File: frontend/src/App.js [hunk 1]
```diff
@@ -12,25 +12,32 @@ class App extends Component {
   state = {
     sources: [],
     err: null,
+    version: {},
   }
 
   componentDidMount() {
     this.loadSources()
+    this.dbVersion()
```
Comment (Harjot1Singh) [inline frontend/src/App.js:20]: Change to something more descriptive. I'm not sure what this does by looking at this line

File: backend/db.js [hunk 2]
```diff
@@ -52,6 +52,22 @@ export const isLatestDatabase = async () => {
...
 
+/**
+ * Get local database version.
+ * @async
+ * @returns {string} Local database verison.
+ */
+export const getDbVersion = async () => {
+  // Read package.json database semver and database package file
+  const [ , { version } ] = await Promise.all( [
+    manifest( databasePackage ),
```
Comment (Harjot1Singh) [inline backend/db.js:63]: What does this do? Read the database package's json? Why is there a readJSON below it?

File: frontend/src/components/Home.js [hunk 3]
```diff
@@ -10,11 +10,12 @@ import './Home.css'
 
 const getPosition = ( source, positions ) => positions[ source ] || { line: 0, page: 1 }
 
-const Home = ( { err, sources, positions } ) => (
+const Home = ( { err, sources, positions, version: { version } } ) => (
```
Comment (Harjot1Singh) [inline frontend/src/components/Home.js:13]: Why is version an object containing version? It'll be better to simply pass in the string. Perhaps the server is returning an object, so it's best to resolve it there.

File: backend/db.js (hunk 2 above)
Comment (saihaj) [reply backend/db.js:63]: It reads the DB version from node_modules

Comment (Harjot1Singh) [reply backend/db.js]: What does the line below do/why are there two lines from reading database package? Did you source the code from Desktop?

File: frontend/src/App.js [hunk 4]
```diff
@@ -12,25 +12,32 @@ class App extends Component {
   state = {
     sources: [],
     err: null,
+    version: '',
   }
 
   componentDidMount() {
     this.loadSources()
+    this.localDbVersion()
```
Comment (Harjot1Singh) [inline frontend/src/App.js:20]: Still the same, it doesn't describe an action, `loadDbVersion()` would be more appropriate.

File: backend/db.js (hunk 2 above)
Comment (Harjot1Singh) [reply backend/db.js]: Oh, I think one must've been for the npm DB version

Comment (saihaj) [reply backend/db.js]: I fixed it now.

File: frontend/src/App.js (hunk 4 above)
Comment (saihaj) [reply frontend/src/App.js:20]: confused?

Comment (Harjot1Singh) [reply frontend/src/App.js:20]: No, since it doesn't describe an action, it simply reads `local db version`

Comment (saihaj) [reply frontend/src/App.js:20]: How does `getLocalDbVersion()` sound?

Comment (Harjot1Singh) [reply frontend/src/App.js:20]: just `getDbVersion()` is ok too, or `loadDbVersion` (inline with `loadSources()`). I prefer the latter for the aforementioned reason.

**Pull request merged**